mnefun/_mnefun.py

```diff
@@ -1368,10 +1369,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
+            flat = p.auto_bad if type(p.auto_bad_flat) == dict else None
```

Comment:
Did you mean `flat = `p.auto_bad_flat if isinstance(p.auto_bad_flat, dict) else None`? I'm still not quite sure on the logic. Right now you are setting it to `flat = p.auto_bad` sometimes, which is either a float or `None`.